src/edge_kernels/VOFAdvectionEdgeAlg.C

```diff
@@ -84,108 +144,112 @@ VOFAdvectionEdgeAlg::execute()
         av[d] = edgeAreaVec.get(edge, d);
       }
 
-      const DblType mdot = massFlowRate.get(edge, 0);
-
       NALU_ALIGNED DblType densityL = density.get(nodeL, 0);
       NALU_ALIGNED DblType densityR = density.get(nodeR, 0);
+      const DblType rhoIp = 0.5 * densityL + 0.5 * densityR;
 
-      NALU_ALIGNED DblType rhoIp = 0.5 * (densityL + densityR);
+      const DblType vdot = massFlowRate.get(edge, 0) / rhoIp;
 
-      const DblType vdot = mdot / rhoIp;
       const DblType qNp1L = scalarQ.get(nodeL, 0);
       const DblType qNp1R = scalarQ.get(nodeR, 0);
 
-      // Compute extrapolated dq/dx
-      NALU_ALIGNED DblType dqL = 0.0;
-      NALU_ALIGNED DblType dqR = 0.0;
-
-      for (int j = 0; j < ndim; ++j) {
-        const DblType dxj =
-          0.5 * (coordinates.get(nodeR, j) - coordinates.get(nodeL, j));
-        dqL += dxj * dqdx.get(nodeL, j);
-        dqR += dxj * dqdx.get(nodeR, j);
-      }
-
-      NALU_ALIGNED DblType limitL = 1.0;
-      NALU_ALIGNED DblType limitR = 1.0;
-
-      if (useLimiter) {
-        const auto dq = scalarQ.get(nodeR, 0) - scalarQ.get(nodeL, 0);
-        const auto dqML = 4.0 * dqL - dq;
-        const auto dqMR = 4.0 * dqR - dq;
-        limitL = van_leer(dqML, dq, eps);
-        limitR = van_leer(dqMR, dq, eps);
-      }
-
-      // Upwind extrapolation with limiter terms
-      NALU_ALIGNED DblType qIpL;
-      NALU_ALIGNED DblType qIpR;
-      qIpL = scalarQ.get(nodeL, 0) + dqL * hoUpwind * limitL;
-      qIpR = scalarQ.get(nodeR, 0) - dqR * hoUpwind * limitR;
-
       // Advective flux
-      const DblType qIp = 0.5 * (qNp1R + qNp1L); // 2nd order central term
-
-      // Upwinded term
-      const DblType qUpw = (vdot > 0) ? (alphaUpw * qIpL + om_alphaUpw * qIp)
-                                      : (alphaUpw * qIpR + om_alphaUpw * qIp);
+      const DblType qIp = 0.5 * (qNp1R + qNp1L);
 
-      const DblType adv_flux = vdot * qUpw;
+      const DblType adv_flux = vdot * qIp;
       smdata.rhs(0) -= adv_flux;
       smdata.rhs(1) += adv_flux;
 
-      // Left node contribution; upwind terms
-      DblType alhsfac = 0.5 * (vdot + stk::math::abs(vdot)) * alphaUpw;
+      DblType alhsfac = 0.5 * vdot;
       smdata.lhs(0, 0) += alhsfac / relaxFac;
       smdata.lhs(1, 0) -= alhsfac;
 
-      // Right node contribution; upwind terms
-      alhsfac = 0.5 * (vdot - stk::math::abs(vdot)) * alphaUpw;
+      alhsfac = 0.5 * vdot;
       smdata.lhs(1, 1) -= alhsfac / relaxFac;
       smdata.lhs(0, 1) += alhsfac;
 
-      // Compression term
-      const DblType velocity_scale = stk::math::abs(
-        vdot / stk::math::sqrt(av[0] * av[0] + av[1] * av[1] + av[2] * av[2]));
+      // Compression + Diffusion term
+      const DblType velocity_scale = global_max_velocity * 5.0;
```

Comment:
Will you add a comment/reference for the rational of the hard coded `5.0`?

Reply:
I'll add a comment here. It comes from a formula in Jain, 2022